solutions/ha-nva/layer-7-ingress/README.md

```diff
@@ -0,0 +1,106 @@
+|DeployLoadBalancerWeb|String|Selection to deploy Azure Application Gateway for the Web|Yes|
+|_artifactsLocation|string|The base URI where artifacts required by this template are located including a trailing '/'|[https://raw.githubusercontent.com/...](https://raw.githubusercontent.com/mspnp/samples/master/solutions/ha-nva/layer-7-ingress/)|
+|_artifactsLocationSasToken|securestring|The sasToken required to access _artifactsLocation||
+## Architecture details
+
+This deployment will set up a simulated environment representing the architecture above.  Clients from the Internet will reach the application via the Public IP on the Application Gateway.  The Application Gateway will forward the request to one of the available deployed NVA VMs. In a real deployment, the NVA VMs may perform other analysis, but for this sample they just forward the request to the erb tier Load Balancer.  The web tier Load Balancer will forward the request to one of the available web VMs.
+
+### Simulated NVA VMs
+
+The simulated NVA VMs for this sample are CentOS 7 or RedHat 7 virtual machines running Apache httpd configured as a reverse proxy to forward all requests to the back end Load Balancer IP address (10.0.1.4). The script to set up the reverse proxy is at: [scripts/simpleproxy.sh](scripts/simpleproxy.sh).
+
+### Simulated Web VMs
+
+The simulated web VMs for this sample are CentOS 7 or RedHat 7 virtual machines running Apache httpd with an index.html file generated to include the VM instance name from the Azure Instance Metadata Service. The script to set up the web server is at: [scripts/simpleweb.sh](scripts/simpleweb.sh).
+
+## Connecting to the web endpoint
+
+To test the deployment, find the public IP address of the Application Gateway (ag-dmz) and enter it in the address bar of your web browser.  You should see a simple page indicating which back-end web server is providing the response. You can stop any of the NVA or web VMs, and the endpoint solution should automatically route requests through and to available VMs.
+
+## Connecting to deployed VMs with Azure Bastion
+
+An Azure Bastion service is deployed to provide access to the VMs in this architecture. The SSH Key required to connect to the VMs is generated at deployment time and stored in Azure Key Vault using the nested deployment script [nested/storekeyinkeyvault.json](nested/storekeyinkeyvault.json), which was adapted from [Create ssh-keys and store in Key Vault](https://azure.microsoft.com/resources/templates/201-deployment-script-ssh-key-gen/).
+
+In order to access the ssh key from the portal to use Azure Bastion, you will need to add an access policy on the Key Vault and grant yourself ***Secret*** **List** and **Get** permissions on `kv-hanva-{uniquestring}`.
+
+> Note: When connecting to VMs using Azure Bastion from the Azure Portal, you will need to click *Advanced* and enter the SSH Key passphrase that you entered when deploying the solution.
```

Comment:
I've been adding a code of conduct to each readme in the samples repo. We have this at the root readme.md file however since we are loading each sample in the code browser I thought it was a good idea to add to each sample as well.

```
## Code of conduct

This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information, see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
```

Reply:
Added, good suggestion